angular/coding-styles/README.md

```diff
@@ -0,0 +1,157 @@
+# Angular Style Guide (Optional)
+
+<br>
+<br>
+
+**this page is in draft mode. it is optional to follow guidelines on this page.**
+
+<br>
+<br>
+
+## Style Guide
+
+### Base Style Guide
+
+The main programing language in Angular framework included with Typescript, HTML, SCSS. We follow
+
+- [Google style guide for HTML/CSS](https://google.github.io/styleguide/htmlcssguide.html), [Typescript](https://google.github.io/styleguide/tsguide.html)
+- [Kitty Giraudel style guide for SCSS](https://sass-guidelin.es/)
+- [Angular Official for Angular](https://angular.io/guide/styleguide).
+
+The rest of the document describes additions and clarifications below.
+
+### Typescript
+
+#### Indentation
+
+The indentation in typescript we use 2 spaces for every new line. we don't tabs or mix tabs.
+
+#### Return Type
+
+We recommend to assign ***return type*** of method for precision and simple to read the surface of them. Some lint configuration as eslint will have warning with highlight.
+
+```typescript
+/** without return type */
+isAdmin(role: string) {
+  return role === 'admin';
+}
+
+/** recommend */
+isAdmin(role: string): boolean {
+  return role === 'admin';
+}
+```
+
+#### Trailing Commas
+
+If the an array or object literal has written in many line. The last field should be ended with comma.
+
+```typescript
+/** Yes */
+const obj = {
+  a: 'value1',
+  b: 'value1',
+  c: 'value1',
+}
+
+/** No */
+const obj = {
+  a: 'value1',
+  b: 'value1',
+  c: 'value1'
+}
+```
+
+### HTML
+
+#### Indentation
+
+The indentation in HTML, we use 2 spaces for inside element. we don't tabs or mix tabs.
```

Comment:
```suggestion
For the indentation in HTML, we use **2 spaces** for elements. we don't use tabs or spaces mixed with tabs.
```

Reply:
Changes:
... for elements. Do not use tabs or spaces mixed with tabs.